docs/cookbook/extending-product-list.md

```diff
@@ -0,0 +1,127 @@
+
+namespace Shopsys\ShopBundle\Model\Product\View;
+
+use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice;
+use Shopsys\ReadModelBundle\Image\ImageView;
+use Shopsys\ReadModelBundle\Product\Action\ProductActionView;
+use Shopsys\ReadModelBundle\Product\Listed\ListedProductView as BaseListedProductView;
+
+class ListedProductView extends BaseListedProductView
+{
+    /**
+     * @var string|null
+     */
+    protected $brandName;
+
+    /**
+     * @param int $id
+     * @param string $name
+     * @param string|null $brandName
+     * @param string|null $shortDescription
+     * @param string $availability
+     * @param \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice $sellingPrice
+     * @param array $flagIds
+     * @param \Shopsys\ReadModelBundle\Product\Action\ProductActionView $action
+     * @param \Shopsys\ReadModelBundle\Image\ImageView|null $image
+     */
+    public function __construct(
+        int $id,
+        string $name,
+        ?string $brandName,
+        ?string $shortDescription,
+        string $availability,
+        ProductPrice $sellingPrice,
+        array $flagIds,
+        ProductActionView $action,
+        ?ImageView $image
+    ) {
+        parent::__construct($id, $name, $shortDescription, $availability, $sellingPrice, $flagIds, $action, $image);
+
+        $this->brandName = $brandName;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getBrandName(): ?string
+    {
+        return $this->brandName;
+    }
+}
+```
+
+### 2. Extend `ListedProductViewFactory` so it returns the new required data
+
+The class is responsible for creating the view object. We need to ensure that the objects is created with proper brand name. We are able to get the brand name from the product entity, so we just need to overwrite `createFromProduct()` method.  
+
+```php
+declare(strict_types=1);
+
+namespace Shopsys\ShopBundle\Model\Product\View;
+
+use Shopsys\FrameworkBundle\Model\Product\Product;
+use Shopsys\ReadModelBundle\Image\ImageView;
+use Shopsys\ReadModelBundle\Product\Action\ProductActionView;
+use Shopsys\ReadModelBundle\Product\Listed\ListedProductViewFactory as BaseListedProductViewFactory;
+use Shopsys\ReadModelBundle\Product\Listed\ListedProductView;
+
+class ListedProductViewFactory extends BaseListedProductViewFactory
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
+     * @param \Shopsys\ReadModelBundle\Image\ImageView|null $imageView
+     * @param \Shopsys\ReadModelBundle\Product\Action\ProductActionView $productActionView
+     * @return \Shopsys\ReadModelBundle\Product\Listed\ListedProductView
+     */
+    protected function createFromProduct(Product $product, ?ImageView $imageView, ProductActionView $productActionView): ListedProductView
+    {
+        return new ListedProductView(
+            $product->getId(),
+            $product->getName(),
+            $product->getBrand() ? $product->getBrand()->getName() : null,
```

Comment:
Newly added parameter is third even it is nullable, why?

Reply:
dont know :laughing: fixed